### PR TITLE
feat: add plan building

### PR DESCRIPTION
**Pull request by ArneD**

### CODE REVIEW

File: src/BuildingRegistry.Api.BackOffice.Handlers/Building/BusHandler.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                .ComputeHash(Encoding.UTF8.GetBytes((string)command.ToString()))
+                .ToHexString();
+
+            // It is possible we have a GUID collision, check the SHA-512 hash as well to see if it is really the same one.
+            // Do nothing if commandId with contenthash exists
+            if (possibleProcessedCommand.Any() && possibleProcessedCommand.ContainsKey(contentHash))
+                throw new IdempotencyException("Already processed");
+
+            var processedCommand = new ProcessedCommand(commandId.Value, contentHash);
+
```
Comment (lvermeulen) [inline src/BuildingRegistry.Api.BackOffice.Handlers/Building/BusHandler.cs:52]: drop newline

File: src/BuildingRegistry.Api.BackOffice/BuildingRegistry.Api.BackOffice.csproj [hunk 2]
```diff
@@ -36,6 +36,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\BuildingRegistry.Api.BackOffice.Handlers\BuildingRegistry.Api.BackOffice.Handlers.csproj" />
```
Comment (lvermeulen) [inline src/BuildingRegistry.Api.BackOffice/BuildingRegistry.Api.BackOffice.csproj:39]: don't we need a dependency on Abstractions as well?

Comment (ArneD) [reply src/BuildingRegistry.Api.BackOffice/BuildingRegistry.Api.BackOffice.csproj:39]: Not really, it is automatically added (somewhere in .net versions).

**Review by lvermeulen — APPROVED** [review]

File: src/BuildingRegistry.Projections.Wms/BuildingV2/BuildingV2Projections.cs [hunk 3]
```diff
@@ -39,6 +39,21 @@ public BuildingV2Projections(WKBReader wkbReader)
...
+                    Version = message.Message.Provenance.Timestamp
+                };
+
+                SetGeometry(buildingV2, message.Message.ExtendedWkbGeometry, BuildingGeometryMethod.Outlined);
+
+                await context.BuildingsV2.AddAsync(buildingV2, ct);
+            });
         }
 
         private static void SetVersion(BuildingV2 building, Instant provenanceTimestamp)
```
Comment (pgallik) [inline src/BuildingRegistry.Projections.Wms/BuildingV2/BuildingV2Projections.cs:59]: Isn't used

File: test/BuildingRegistry.Tests/AggregateTests/WhenPlanningBuilding/GivenNoBuildingExists.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        {
+            var command = Fixture.Create<PlanBuilding>();
+
+            // Act
+            var result = Building.Plan(
+                command.BuildingPersistentLocalId,
+                command.Geometry);
+
+            // Assert
+            result.Should().NotBeNull();
```
Comment (pgallik) [inline test/BuildingRegistry.Tests/AggregateTests/WhenPlanningBuilding/GivenNoBuildingExists.cs:61]: BuildingStatus?

**Review by pgallik — APPROVED** [review]

File: src/BuildingRegistry.Projections.Wms/BuildingV2/BuildingV2Projections.cs (hunk 3 above)
Comment (ArneD) [reply src/BuildingRegistry.Projections.Wms/BuildingV2/BuildingV2Projections.cs:59]: Correct, for later use.

File: test/BuildingRegistry.Tests/AggregateTests/WhenPlanningBuilding/GivenNoBuildingExists.cs (hunk 4 above)
Comment (ArneD) [reply test/BuildingRegistry.Tests/AggregateTests/WhenPlanningBuilding/GivenNoBuildingExists.cs:61]: Good catch 👍

**Pull request merged**